src/common/components/clusterConfiguration/DownloadIpxeScript.tsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import { saveAs } from 'file-saver';
+import {
+  Alert,
+  Button,
+  ButtonVariant,
+  ClipboardCopy,
+  clipboardCopyFunc,
+  ModalBoxBody,
+  ModalBoxFooter,
+  Stack,
+  StackItem,
+} from '@patternfly/react-core';
+import { DetailItem, DetailList } from '../ui';
+import DiscoveryIpxeInstructions from './DiscoveryIpxeInstructions';
+import { useTranslation } from '../../hooks/use-translation-wrapper';
+
+export type DownloadISOProps = {
+  isSNO?: boolean;
+  fileName?: string;
+  downloadUrl?: string;
+  onClose: () => void;
+  onReset?: () => void;
+};
+
+const DownloadIpxeScript = ({
+  fileName = 'ipxe-script.txt',
+  downloadUrl,
+  onClose,
+  onReset,
+}: DownloadISOProps) => {
+  const wgetCommand = `wget -O ${fileName} '${downloadUrl || ''}'`;
+  const { t } = useTranslation();
+  return (
+    <>
+      <ModalBoxBody>
+        <Stack hasGutter>
+          <StackItem>
+            <Alert
+              isInline
+              variant="success"
+              title={t('ai:iPXE script file is ready to be downloaded')}
+            >
+              <>
+                &nbsp;
+                <DiscoveryIpxeInstructions />
+              </>
+            </Alert>
+          </StackItem>
+          <StackItem>
+            <DetailList>
+              <DetailItem
+                title={t('ai:iPXE script file URL')}
+                value={
+                  <ClipboardCopy
+                    isReadOnly
+                    onCopy={(event) => clipboardCopyFunc(event, downloadUrl)}
+                  >
+                    {downloadUrl}
+                  </ClipboardCopy>
+                }
+              />
+              <DetailItem
+                title={t('ai:Command to download the iPXE script file')}
+                value={
+                  <ClipboardCopy
+                    isReadOnly
+                    onCopy={(event) => clipboardCopyFunc(event, wgetCommand)}
+                  >
+                    {wgetCommand}
+                  </ClipboardCopy>
+                }
+              />
+            </DetailList>
+          </StackItem>
+        </Stack>
+      </ModalBoxBody>
+      <ModalBoxFooter>
+        <Button
+          variant={ButtonVariant.primary}
+          onClick={() => downloadUrl && saveAs(downloadUrl)}
```

Comment:
- I tried to download the file by clicking on the button and I got an error page. Shouldn't it work?

- I then copied the URL in a separate window, and a file was downloaded with the name `download` but looks like it's a zip so IMO the extension should be part of the name. Maybe this is handled on the API?

Reply:
The URL in localhost needs to be changed:
http://10.1.178.17:6000/api/assisted-install/v2/infra-envs/a733ea24-6b1e-495f-a773-bcea1d93f5f0/downloads/files?file_name=ipxe-script
to